From 68509c84a9e6d038928882e1e76a518d4a2f9513 Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Thu, 2 Apr 2020 07:50:35 +0000
Subject: [PATCH] Fix compiling warnings of file arm64/kernel/perf_callchain.c

arch/arm64/kernel/perf_callchain.c:107:6: warning: cast from
pointer to integer of different size [-Wpointer-to-int-cast]
 if ((u32)tail + 4 >= buftail.fp)
      ^
arch/arm64/kernel/perf_callchain.c:110:9: warning: cast to
pointer from integer of different size [-Wint-to-pointer-cast]
  return (struct compat_frame_tail __user *)(buftail.fp - 4);
         ^

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 arch/arm64/kernel/perf_callchain.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/kernel/perf_callchain.c b/arch/arm64/kernel/perf_callchain.c
index b9bbe0b04fc1..c054d404bebd 100644
--- a/arch/arm64/kernel/perf_callchain.c
+++ b/arch/arm64/kernel/perf_callchain.c
@@ -97,10 +97,11 @@ compat_user_backtrace(struct compat_frame_tail __user *tail,
 	 * Frame pointers should strictly progress back up the stack
 	 * (towards higher addresses).
 	 */
-	if ((u32)tail + 4 >= buftail.fp)
+	if ((u32)(u64)tail + 4 >= buftail.fp)
 		return NULL;
 
-	return (struct compat_frame_tail __user *)(buftail.fp - 4);
+
+	return (struct compat_frame_tail __user *)((u64)buftail.fp - 4);
 }
 #endif /* CONFIG_COMPAT */
 
-- 
2.17.1

